### ISSUE TITLE
cpanm hangs indefinitely

### ISSUE THREAD

**Issue by jkeenan** [miyagawa/cpanminus#526]
I'm reporting this problem even though it appears to have cleared itself up -- for now.

I recently used ```perlbrew``` to install perl-5.24.1.  I installed App::cpanminus against that perl.
```
$ which perl
/home/jkeenan/perl5/perlbrew/perls/perl-5.24.1/bin/perl
$ which cpanm
/home/jkeenan/perl5/perlbrew/perls/perl-5.24.1/bin/cpanm
$ ls -l /home/jkeenan/perl5/perlbrew/perls/perl-5.24.1/bin/perl /home/jkeenan/perl5/perlbrew/perls/perl-5.24.1/bin/cpanm
-r-xr-xr-x 1 jkeenan jkeenan  304916 Jan 14 21:05 /home/jkeenan/perl5/perlbrew/perls/perl-5.24.1/bin/cpanm
-rwxr-xr-x 2 jkeenan jkeenan 1945560 Jan 14 21:02 /home/jkeenan/perl5/perlbrew/perls/perl-5.24.1/bin/perl
```
For various projects I then used ```cpanm``` to install or upgrade lots of modules.  They installed without incident and without any visible difference from other instances of using ```cpanm``` over recent years.

Tonight I said:
```
cpanm Email::Stuffer
```
The program went through its normal paces, ending up by saying that Email::Stuffer had been installed successfully -- and then hung indefinitely.  Minutes went by without my getting my command-line prompt back.  Checking ```ps aux```, I was told that ```cpanm``` was in Interruptible Sleep state (D+).  I eventually killed the process via ```Ctrl-C```.

I then tried ```cpanm List::Compare``` -- a module which I knew was both installed and up-to-date.
```
$ cpanm List::Compare
List::Compare is up to date. (0.53)

```
Again, minutes went by without the program exiting cleanly; again I had to ```Ctrl-C```.

This problem persisted after (a) logging out and logging back in; and then (b) a complete restart (Ubuntu Linux 16.04).  Eventually I got the idea of timing the process:
```
$ time cpanm List::Compare
List::Compare is up to date. (0.53)

real	2m28.515s
user	0m2.972s
sys	0m3.412s
```
So at last ```cpanm``` was exiting cleanly -- but taking a vastly excessive time to do so!

After this invocation, ```cpanm``` started to run at a normal speed .
```
$ time cpanm Parse::Taxonomy
Parse::Taxonomy is up to date. (0.24)

real	0m0.443s
user	0m0.148s
sys	0m0.012s

$ time cpanm List::Compare
List::Compare is up to date. (0.53)

real	0m0.398s
user	0m0.140s
sys	0m0.024s

$ time cpanm ExtUtils::ModuleMaker
ExtUtils::ModuleMaker is up to date. (0.55)

real	0m0.384s
user	0m0.164s
sys	0m0.024s
```
Has anyone else observed this problem?  Does anyone know what can cause this hanging?

Thank you very much.

Jim Keenan

**Comment by miyagawa** [comment]
It might be the code that runs at the end of cpanm that removes stale
working directories.

**Comment by jkeenan** [comment]
Hmm, that's a real possibility.

I have been working on https://rt.perl.org/Ticket/Display.html?id=130497 and trying to identify CPAN distributions which are likely to be impacted by the fatalization in perl-5.26.0 of one currently deprecated feature.  There is one distribution where this "blead-breaks-CPAN" problem is particularly critical because thousands of other CPAN distributions depend on this module.  So I:

* Created a list of all the CPAN distributions which depended on the critical distribution.

* Installed perl 5 blead and installed ```cpanm``` against that ```perl```.

* Used ```cpanm``` to try to install a *corrected version* of the critical distribution.

* Wrote a program in which I iterated over the list mentioned above and used ```cpanm``` to try to install each distribution therein against blead perl.

Since I had to stop and re-start that program several times, it's likely that I had created tens of thousands of temporary directories which would not have been removed until reaching some expiration date set by code within ```cpanm```.  It is plausible that yesterday was the expiration date and that each of my successive invocations of ```cpanm``` was removing a portion of that backlog.

Perhaps some advice about this could be included in the documentation?

Thank you very much.

Jim Keenan
